robots/MINDCUB3R/mindcuber.py

```diff
@@ -148,20 +159,18 @@ def rotate_cube_blocked(self, direction, nb):
                  (direction, nb, current_pos, temp_pos, final_pos))
 
         self.turntable.run_to_abs_pos(position_sp=temp_pos,
-                                      speed_sp=Rubiks.rotate_speed,
+                                      speed_sp=MindCuber.rotate_speed,
                                       stop_action='hold',
                                       ramp_up_sp=0)
-        self.turntable.wait_for_running()
-        self.turntable.wait_for_position(temp_pos)
-        self.turntable.wait_for_stop()
+        self.turntable.wait_until('running')
+        self.turntable.wait_until_not_moving()
 
         self.turntable.run_to_abs_pos(position_sp=final_pos,
-                                      speed_sp=int(Rubiks.rotate_speed/4),
+                                      speed_sp=int(MindCuber.rotate_speed/4),
                                       stop_action='hold',
                                       ramp_up_sp=0)
-        self.turntable.wait_for_running()
-        self.turntable.wait_for_position(final_pos, stall_ok=True)
-        self.turntable.wait_for_stop()
+        self.turntable.wait_until('running')
+        self.turntable.wait_until_not_moving()
```

Comment:
I rebuilt MINDCUBER and solved several cubes to test all of the changes.  The `wait_until('running')` followed by `wait_until_not_moving()` combination works extremely well.

Reply:
Should we make a helper for that? It seems like you need to do this pretty much every time.

Also, I think we should pay attention to the new APIs; where possible,. I would prefer that we update our demo code do use them rather than the older ones.

Reply:
Good point I hadn't thought about switching over to the `on_for_XYZ` calls...will do that.  All of the `on_for_XYZ` methods do `wait_until('running')` followed by `wait_until_not_moving()` (assuming `block` is `True`) so that will clean up most of these.